src/models/__tests__/community_business.test.integration.ts

```diff
@@ -296,4 +297,50 @@ describe('Community Business Model', () => {
       }
     });
   });
+
+
+  describe('byRegion', () => {
+    test(':: returns all Community Businesses names by region ', async () => {
+      const cbByRegion = await CommunityBusinesses.byRegion(trx, {});
+      expect(cbByRegion).toEqual({
+        London: [
+          { id: 1, name: 'Aperture Science' },
+          { id: 2, name: 'Black Mesa Research' },
+        ]});
+    });
+
+    test(':: returns subset of Community Businesses names by region with query', async () => {
+      const blackMesa = await CommunityBusinesses
+        .getOne(trx, { where: { name: 'Black Mesa Research' } });
+      await CommunityBusinesses.destroy(trx, blackMesa);
+
+      const cbByRegion = await CommunityBusinesses.byRegion(trx, { where: { deletedAt: null } });
+      expect(cbByRegion).toEqual({ London: [{ id: 1, name: 'Aperture Science' }] });
+    });
+
+    test(':: returns alphebatised list', async () => {
+      await CommunityBusinesses.add(trx, {
+        name: 'Zeldas Dungeon',
+        region: RegionEnum.LONDON,
+        sector: SectorEnum.COMMUNITY_HUB,
+      });
+      await CommunityBusinesses.add(trx, {
+        name: 'Okamis Fields',
+        region: RegionEnum.LONDON,
+        sector: SectorEnum.ENVIRONMENT,
+      });
+      /*
+       * NB: due to sequences not resetting during transaction rollbacks this test
```

Comment:
Then just don't assert against the IDs. Use `expect.objectContaining`.

Reply:
```ts
expect(cbByRegion).toEqual({
  London: [
    expect.objectContaining({ name: 'Aperture Science' }),
    expect.objectContaining({ name: 'Black Mesa Research' }),
    expect.objectContaining({ name: 'Okamis Fields' }),
    expect.objectContaining({ name: 'Zeldas Dungeon' }),
  ]
});
```

Doesn't seem like much extra. Or even:
```ts
expect(cbByRegion).toEqual({
  London: [
    { name: 'Aperture Science' },
    { name: 'Black Mesa Research' },
    { name: 'Okamis Fields' },
    { name: 'Zeldas Dungeon' },
  ].map(expect.objectContaining)
});
```

Reply:
@eliascodes I know how to implement what you were suggesting. I wrote this test as it is because it explicitly shows that the results are ordered in alphabetically rather than by id. It is also gives a more readable return value for when you are looking back at tests.

As this is test code, not application code, I do not think we need to worry about the small case when one of us wants to run this test in isolation.